webapp/src/pages/invitedGuests/components/InvitedGuestComponent.js

```diff
@@ -152,25 +152,16 @@ class InvitedGuests extends Component {
     }
   };
 
+
+
   filterByName = field => {
     let searchList = this.state.guestList;
 
     let value = field.target.value.toLowerCase();
     var roleSearch = this.state.roleSearch;
     let tempList = searchList.filter(  (guest) =>  {
       let fullname = guest.user.user_title + " " + guest.user.firstname + " " + guest.user.lastname;
-
-      if (fullname.toLowerCase().indexOf(value) > -1)
-        if(roleSearch != "all" )
-        {
-          if( guest.role === roleSearch)
-          {
-            return guest;
-          }
-        }
-        else{
-          return guest;
-        }
+      return (fullname.toLowerCase().indexOf(value) > -1) && (roleSearch === "all" || guest.role === roleSearch)
```

Comment:
This part needs to return a guest, this way it would only return a true/false value.

Reply:
The callback function in filter is for testing if the element should be included or not
From mozila

> filter() calls a provided callback function once for each element in an array and constructs a new array of all the values for which callback returns a value that coerces to true